src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const property of action.keysToRemove) {
+          if (Object.keys(state).filter(elem => elem === property)) {
+            delete state[property];
+          }
+        }
+        break;
+
+      case 'clear':
+        for (const key of Object.keys(state)) {
+          delete state[key];
+        }
```

Comment:
```suggestion
      case 'clear':
        for (const key in state) {
          delete state[key];
        }
```

Reply:
checklist has a rule 'don't use for in loop for iterating over array'